python/tank/authentication/ui_authentication.py

```diff
@@ -32,14 +31,17 @@ class UiAuthenticationHandler(object):
     directly and be used through the authenticate and renew_session methods.
     """
 
-    def __init__(self, is_session_renewal, fixed_host=False):
+    def __init__(self, is_session_renewal, fixed_host=False, cookies=None):
         """
         Creates the UiAuthenticationHandler object.
         :param is_session_renewal: Boolean indicating if we are renewing a session. True if we are, False otherwise.
+        :param fixed_host: Indicate if the user can select a different host for connecting to.
+        :param cookies: String of raw cookies
         """
         self._is_session_renewal = is_session_renewal
         self._gui_launcher = invoker.create()
         self._fixed_host = fixed_host
+        self._cookies = cookies
```

Comment:
A user object can exist only if you are authenticated. This code path is used both for authenticated and unauthenticated scenarios.